servo/connectors/kubernetes.py

```diff
@@ -2003,6 +2016,578 @@ async def get_restart_count(self) -> int:
 
         return count
 
+# Workarounds to allow use of api_client.deserialize() public method instead of private api_client._ApiClient__deserialize
+# TODO: is this workaround worth it just to avoid using the private method?
+# fix for https://github.com/kubernetes-client/python/issues/977#issuecomment-594045477
+def default_kubernetes_json_serializer(o: Any) -> Any:
+    if isinstance(o, (datetime.datetime, datetime.date)):
+        return o.isoformat()
+    raise TypeError(f'Object of type {o.__class__.__name__} '
+                        f'is not JSON serializable')
+
+# https://github.com/kubernetes-client/python/issues/977#issuecomment-592030030
+class FakeKubeResponse:
+    """Mocks the RESTResponse object as a workaround for kubernetes python api_client deserialization"""
+    def __init__(self, obj):
+        self.data = json.dumps(obj, default=default_kubernetes_json_serializer)
+
+# Use alias generator so that lower camel case can be parsed to snake case properties to match k8s python client behaviour
+def to_lower_camel(string: str) -> str:
+    split = string.split('_')
+    return split[0] + ''.join(word.capitalize() for word in split[1:])
+
+class RolloutBaseModel(pydantic.BaseModel):
+    class Config:
+        # arbitrary_types_allowed = True
+        alias_generator = to_lower_camel
+        allow_population_by_field_name = True
+
+# Pydantic type models for argo rollout spec: https://argoproj.github.io/argo-rollouts/features/specification/
+# https://github.com/argoproj/argo-rollouts/blob/master/manifests/crds/rollout-crd.yaml
+# NOTE/TODO: fields typed with Any should maintain the same form when dumped as when they are parsed. Should the need
+#   arise to interact with such fields, they will need to have an explicit type defined so the alias_generator is applied
+class RolloutV1LabelSelector(RolloutBaseModel): # must type out k8s models as well to allow parse_obj to work
+    match_expressions: Any
```

Comment:
We can do better than this on the type identifier, no?

Reply:
https://github.com/opsani/servox/pull/303#discussion_r685491402